.github/workflows/tests.yml

```diff
@@ -0,0 +1,156 @@
+name: Build tests
+
+on:
+  push:
+    branches: [ $default-branch ]
+  pull_request:
+    branches: [ $default-branch ]
```

Comment:
I think limiting the branches here disables CI runs for future pull requests.

Reply:
My understanding is that it limits to PRs to the default branch and pushes to the default branch:

https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags

It is also used in the example workflows:

https://github.com/actions/starter-workflows/blob/main/ci/python-package.yml
https://github.com/actions/starter-workflows/blob/main/ci/c-cpp.yml